docs/guides/payments.rst

```diff
@@ -0,0 +1,236 @@
+.. _adding-payments:
+
+Payments
+========
+
+Integrating a new Payment Gateway into Saleor
+---------------------------------------------
+
+We are using a universal flow, that each provider should fulfill, there are
+several methods that should be implemented.
+
+Your changes should live under the
+``saleor.payment.providers.<provider name>`` module.
+
+.. note::
+
+    After completing those steps your new payment gateway will only be
+    available from the API level. You will also need to integrate it into your
+    Frontend's workflow.
+
+get_transaction_token(**connection_params)
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+A transaction token is a signed data blob that includes configuration and
+authorization information required by the payment provider.
+
+These should not be reused; a new transaction token should be generated for
+each payment request.
+
+Example
+"""""""
+
+.. code-block:: python
+
+    def get_transaction_token(**connection_params: Dict) -> str:
+        gateway = get_gateway(**connection_params)
+        transaction_token = gateway.transaction_token.generate()
+        return transaction_token
+
+authorize(payment, transaction_token, **connection_params)
```

Comment:
Over the docs, there are multiple warnings about put Python kwargs being empty bold instructions. Which also produces what seems to be an unwanted rendering.

The below patch fixes that.
```patch
Index: docs/guides/payments.rst
<+>UTF-8
===================================================================
--- docs/guides/payments.rst	(revision a4f77a9d22216905738c6442867871f0b48ea1fb)
+++ docs/guides/payments.rst	(date 1540673610000)
@@ -18,8 +18,8 @@
     available from the API level. You will also need to integrate it into your
     Frontend's workflow.
 
-get_transaction_token(**connection_params)
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+get_transaction_token(\*\*connection_params)
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
 A transaction token is a signed data blob that includes configuration and
 authorization information required by the payment provider.
@@ -37,8 +37,8 @@
         transaction_token = gateway.transaction_token.generate()
         return transaction_token
 
-authorize(payment, transaction_token, **connection_params)
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+authorize(payment, transaction_token, \*\*connection_params)
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
 A process of reserving the amount of money against the customer's funding
 source. Money does not change hands until the authorization is captured.
@@ -67,8 +67,8 @@
         return txn, response['error']
 
 
-refund(payment, amount, **connection_params)
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+refund(payment, amount, \*\*connection_params)
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
 Full or partial return of captured funds to the customer.
 
@@ -100,8 +100,8 @@
             gateway_response=get_gateway_response(response))
         return txn, response['error']
 
-capture(payment, amount, **connection_params)
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+capture(payment, amount, \*\*connection_params)
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
 A transfer of the money that was reserved during the authorization stage.
 
@@ -133,8 +133,8 @@
             gateway_response=get_gateway_response(response))
         return txn, response['error']
 
-void(payment, **connection_params)
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+void(payment, \*\*connection_params)
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
 A cancellation of a pending authorization or capture.
```

Reply:
Good point, thanks